mmseg/datasets/custom.py

```diff
@@ -164,13 +164,14 @@ def load_annotations(self, img_dir, img_suffix, ann_dir, seg_map_suffix,
                 img_infos.append(img_info)
         else:
             for img in self.file_client.list_dir_or_file(
-                    dir_path=img_dir,
+                    dir_path=ann_dir,
```

Comment:
how do we handle `ann_dir=None`?

Reply:
if need handle ann_dir = None
can：
assert ann_dir==None: 'please add anntations file!'
Is this possible?

Reply:
This PR is just some suggestions from me, if you think it is not necessary at all, you can just close it

Reply:
I think it would be better to still get image path first, and get seg map path by replace the suffix and check whether seg map exists by

```python
if self.file_client(seg_map):
    img_info['ann'] = dict(seg_map=seg_map)
else:
    continue
```
what do you think about this modification?